N802 false positives for `CGIHTTPRequestHandler` and `SimpleHTTPRequestHandler`

### Summary

#18907 added an exception to [`invalid-function-name` (N802)](https://docs.astral.sh/ruff/rules/invalid-function-name/) for `http.server.BaseHTTPRequestHandler`. That exception should also apply to its subclasses, `http.server.CGIHTTPRequestHandler` and `http.server.SimpleHTTPRequestHandler`. [Example](https://play.ruff.rs/5f94732e-948d-4404-9530-4e7ffffe8706):
```console
$ cat >n802.py <<'# EOF'
from http.server import CGIHTTPRequestHandler, SimpleHTTPRequestHandler
class C(CGIHTTPRequestHandler):
    def do_OPTIONS(self):
        pass
class S(SimpleHTTPRequestHandler):
    def do_OPTIONS(self):
        pass
# EOF

$ ruff --isolated check n802.py --select N802 --output-format concise -q
n802.py:3:9: N802 Function name `do_OPTIONS` should be lowercase
n802.py:6:9: N802 Function name `do_OPTIONS` should be lowercase
```

### Version

ruff 0.12.4 (ee2759b36 2025-07-17)

--- a/crates/ruff_linter/resources/test/fixtures/pep8_naming/N802.py
+++ b/crates/ruff_linter/resources/test/fixtures/pep8_naming/N802.py
@@ -84,3 +84,25 @@ class MyRequestHandler(BaseHTTPRequestHandler):
     def dont_GET(self):
         pass
 
+
+from http.server import CGIHTTPRequestHandler
+
+
+class MyCGIRequestHandler(CGIHTTPRequestHandler):
+    def do_OPTIONS(self):
+        pass
+
+    def dont_OPTIONS(self):
+        pass
+
+
+from http.server import SimpleHTTPRequestHandler
+
+
+class MySimpleRequestHandler(SimpleHTTPRequestHandler):
+    def do_OPTIONS(self):
+        pass
+
+    def dont_OPTIONS(self):
+        pass
+

--- a/crates/ruff_linter/src/rules/pep8_naming/rules/invalid_function_name.rs
+++ b/crates/ruff_linter/src/rules/pep8_naming/rules/invalid_function_name.rs
@@ -100,16 +100,16 @@ pub(crate) fn invalid_function_name(
         return;
     }
 
-    // Ignore the do_* methods of the http.server.BaseHTTPRequestHandler class
+    // Ignore the do_* methods of the http.server.BaseHTTPRequestHandler class and its subclasses
     if name.starts_with("do_")
         && parent_class.is_some_and(|class| {
             any_base_class(class, semantic, &mut |superclass| {
                 let qualified = semantic.resolve_qualified_name(superclass);
                 qualified.is_some_and(|name| {
-                    matches!(
-                        name.segments(),
-                        ["http", "server", "BaseHTTPRequestHandler"]
-                    )
+                    let segments = name.segments();
+                    segments == ["http", "server", "BaseHTTPRequestHandler"]
+                        || segments == ["http", "server", "CGIHTTPRequestHandler"]
+                        || segments == ["http", "server", "SimpleHTTPRequestHandler"]
                 })
             })
         })

--- a/crates/ruff_linter/src/rules/pep8_naming/rules/invalid_function_name.rs
+++ b/crates/ruff_linter/src/rules/pep8_naming/rules/invalid_function_name.rs
@@ -106,10 +106,16 @@ pub(crate) fn invalid_function_name(
             any_base_class(class, semantic, &mut |superclass| {
                 let qualified = semantic.resolve_qualified_name(superclass);
                 qualified.is_some_and(|name| {
-                    let segments = name.segments();
-                    segments == ["http", "server", "BaseHTTPRequestHandler"]
-                        || segments == ["http", "server", "CGIHTTPRequestHandler"]
-                        || segments == ["http", "server", "SimpleHTTPRequestHandler"]
+                    matches!(
+                        name.segments(),
+                        [
+                            "http",
+                            "server",
+                            "BaseHTTPRequestHandler"
+                                | "CGIHTTPRequestHandler"
+                                | "SimpleHTTPRequestHandler"
+                        ]
+                    )
                 })
             })
         })

